code/datums/station_traits/neutral_traits.dm

```diff
@@ -189,3 +189,101 @@
 		return
 
 	gorilla.poll_for_gorilla()
+
+/datum/station_trait/birthday
+	name = "Employee Birthday"
+	trait_type = STATION_TRAIT_NEUTRAL
+	weight = 2
+	show_in_report = TRUE
+	report_message = ""
+	trait_to_give = STATION_TRAIT_BIRTHDAY
+	blacklist = list(/datum/centcom_announcer/intern, /datum/station_trait/announcement_medbot)
+	var/mob/living/carbon/human/birthday_person
+
+/datum/station_trait/birthday/New()
+	. = ..()
+	RegisterSignals(SSdcs, list(COMSIG_GLOB_JOB_AFTER_SPAWN), PROC_REF(on_job_after_spawn))
+
+/datum/station_trait/birthday/revert()
+	for (var/obj/effect/landmark/start/hangover/party_spot in GLOB.start_landmarks_list)
+		QDEL_LIST(party_spot.party_debris)
+	return ..()
+
+/datum/station_trait/birthday/on_round_start()
+	. = ..()
+	var/list/birthday_options
+	for(var/mob/living/carbon/human/human in GLOB.human_list)
+		if(human.mind?.assigned_role.job_flags & JOB_CREW_MEMBER)
+			birthday_options += human
+	birthday_person = pick(birthday_options)
+	addtimer(CALLBACK(src, PROC_REF(announce_birthday)), 10 SECONDS)
+
+/datum/station_trait/birthday/proc/announce_birthday()
+	priority_announce("Happy Birthday to [birthday_person ? birthday_person.name : "Employee Name"]! we hope you enjoy your [birthday_person ? thtotext(birthday_person.age) : "255th"] Birthday")
```

Comment:
I was initially thinking of having a document for a bunch of different variations of this that it would then pick from, but looking at this and how it's done I realize now that would be a very difficult task.